### PR TITLE
Coveralls + AppVeyor + Travis CI

### PR DESCRIPTION
**Pull request by anthonydresser**
Enables the three service for this repo.

TODO:
- Once the appveyor account is created, modify the readme to point to the correct url
- Once the appveyor and coveralls accounts are created, get the repo token from coveralls (should be on front page) and encrypt it using appveyor. Then take that encrypted token and put it in the appveyor.xml where it says Insert coveralls token here
- Enable integration tests to bring code coverage up

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 417A0893;
+      sudo apt-get update;
+      sudo apt-get install dotnet-dev-1.0.0-preview2.1-003177;
+    else
+      brew update;
+      brew install openssl;
+      mkdir -p /usr/local/lib;
+      ln -s /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib /usr/local/lib/;
+      ln -s /usr/local/opt/openssl/lib/libssl.1.0.0.dylib /usr/local/lib/;
+      brew cask install dotnet;
```
Comment (kevcunnane) [inline .travis.yml]: Do we need a version here? Will this start installing dotnet 1.1 which we know can have issues?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      export PATH="/usr/local/share/dotnet/:$PATH";
+    fi
+install:
+  - dotnet restore
+
+script:
+  - dotnet build src/Microsoft.SqlTools.ServiceLayer
+  - dotnet test test/Microsoft.SqlTools.ServiceLayer.Test
```
Comment (kevcunnane) [inline .travis.yml]: Minor: this warns of no newline at end of file. Should fix?

File: README.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+[![Travis CI](https://travis-ci.org/Microsoft/sqltoolsservice.svg?branch=dev)](https://travis-ci.org/anthonydresser/sqltoolsservice)
```
Comment (kevcunnane) [inline README.md]: We'll change this to https://travis-ci.org/**Microsoft**/sqltoolsservice once this is checked in / created, correct?

File: .travis.yml (hunk 2 above)
Comment (anthonydresser) [inline .travis.yml]: what exactly was complaining? I added it anyways.

Comment (kevcunnane) [inline .travis.yml]: It showed up as a warning in the GitHub code review itself :) Like I said, pretty minor. Some Unix-based things have issues without a newline thought

File: .travis.yml (hunk 1 above)
Comment (kevcunnane) [inline .travis.yml]: Sounds good, thanks for the clarification

**Pull request merged**